File: src/main/java/com/gdosoftware/biblioteca/domain/casousos/SocioCasoUsoImpl.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.gdosoftware.biblioteca.domain.casousos;

import com.gdosoftware.biblioteca.domain.interfaces.ISocioCasoUso;
import com.gdosoftware.biblioteca.domain.interfaces.ISocioRepository;
import com.gdosoftware.biblioteca.domain.modelo.Socio;
import java.util.List;

/**
 *
 * @author Dani
 */
public class SocioCasoUsoImpl implements ISocioCasoUso{
    
    private ISocioRepository socioRepo;

    @Override
    public Socio createLibro(Socio socio) {
        return socioRepo.create(socio);
    }

    @Override
    public Socio retriveLibro(Long id) {
        return socioRepo.retrieve(id);
    }

    @Override
    public Socio updateLibro(Long id, Socio socio) {
        return socioRepo.update(id, socio);
    }

    @Override
    public void deleteLibro(Long id) {
        socioRepo.deltete(id);
    }

    @Override
    public List<Socio> findAllLibro() {
        return socioRepo.findAll();
    }
    
}

File: domain/casousos/socioCasoUsoImpl.go
package casousos

import (
	"github.com/gdosoftware/biblioteca/domain/interfaces"
	"github.com/gdosoftware/biblioteca/domain/modelo"
)

type SocioCasoUsoImpl struct {
	Repo interfaces.ISocioRepository
}

func CreateSocioCasoUsoImpl(repo interfaces.ISocioRepository) *SocioCasoUsoImpl{
	return &SocioCasoUsoImpl{Repo:repo}
}

func (r *SocioCasoUsoImpl) CreateSocio(socio *modelo.Socio) (*modelo.Socio, error) {
	return r.Repo.Create(socio)
}

func (r *SocioCasoUsoImpl) RetrieveSocio(id string) (modelo.Socio, error) {
	return r.Repo.Retrieve(id)
}

func (r *SocioCasoUsoImpl) UpdateSocio(id string, socio *modelo.Socio) (*modelo.Socio, error) {
	return r.Repo.Update(id, socio)
}

func (r *SocioCasoUsoImpl) DeleteSocio(id string) error {
	return r.Repo.Delete(id)
}

func (r *SocioCasoUsoImpl) FindAllSocio() ([]modelo.Socio, error) {
	return r.Repo.FindAll()
}
